pkg/backupdriver/backup_driver_controller_base.go

```diff
@@ -493,12 +504,22 @@ func (ctrl *backupDriverController) syncCloneFromSnapshotByKey(key string) error
 
 	switch phase := cloneFromSnapshot.Status.Phase; phase {
 	case backupdriverapi.ClonePhaseInProgress:
-	case backupdriverapi.ClonePhaseCompleted:
 	case backupdriverapi.ClonePhaseFailed:
 	case backupdriverapi.ClonePhaseCanceling:
 	case backupdriverapi.ClonePhaseCanceled:
 		ctrl.logger.Infof("Skipping cloneFromSnapshot %s which is not in New or Retry phase. Current phase: %v", key, cloneFromSnapshot.Status.Phase)
 		return nil
+	case backupdriverapi.ClonePhaseCompleted:
+		// If Snapshot CR status reaches terminal state more than 1 hour, Snapshot CR should be deleted
+		now := time.Now()
+		if cloneFromSnapshot.Status.CompletionTimestamp.Add(time.Hour).Unix() < now.Unix() {
```

Comment:
Same here.  The time period should be configurable, not hard-coded.

Reply:
Same as above.